### DISCUSSION TITLE
KeyedTuple.__allow_access_to_unprotected_subobjects__  attribute

### DISCUSSION THREAD

**Discussion by soujanyat** [sqlalchemy/sqlalchemy#10924] [Usage Questions]
Hi,

We are in the process of upgrading sqlalchemy version 2.0.25

We have a scenario where `KeyedTuple.__allow_access_to_unprotected_subobjects__`  attribute is using in our project. Since KeyedTuple is removed and Row the replace of KeyedTuple does not have this attribute so please suggest what is the alternative for this attributes.

Usage:
```
from sqlalchemy.util import KeyedTuple                                                                                         

KeyedTuple.__allow_access_to_unprotected_subobjects__ = 1
```

Thank You,

**Comment by CaselIT** [comment]
Hi,

from which version of sqlalchemy are you migrating from? KeyedTuple was removed in 1.4 IIRC

**Reply by soujanyat** [reply]
From SQLAlchemy = 1.3.17

**Comment by zzzeek** [comment]
that attribute is not part of SQLAlchemy.  If your application has that attribute available on KeyedTuple it would suggest you have some third party extension in place, based on a google I would suggest it's [zope security proxies](https://www.zope.dev/zope_secrets/security.html), that is providing for this feature.  I would reach out to Zope developers / support for background on how to migrate to SQLAlchemy 2.0 which no longer uses this object.
